segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/MetadataStore.java

```diff
@@ -704,8 +704,16 @@ static SegmentInfo newSegment(String name, Collection<AttributeUpdate> attribute
                     .build();
         }
 
+        // createSegment in Metadata uses this method to get SegmentInfo
+        static SegmentInfo recoveredSegment(SegmentProperties segmentProperties) {
+            return builder()
+                    .segmentId(ContainerMetadata.NO_STREAM_SEGMENT_ID)
```

Comment:
This is incorrect. Please make sure you use the code you have in `DebugStreamSegmentContainer` instead of setting this here.

Reply:
Done.